### ISSUE TITLE
Uncaught Error: Permission denied to access property "__hb_original"

### ISSUE THREAD

**Issue by joshuap** [honeybadger-io/honeybadger-js#1001]
## What are the steps to reproduce this issue?
I haven't figured out how to reliably reproduce the issue yet, but it seems to be some combination of honeybadger.js + the [Readwise Highlighter](https://addons.mozilla.org/en-US/firefox/addon/readwise-highlighter/) plugin in Firefox.

## What happens?
The following error is in the console, and in at least one case it prevented the customer's app from loading:

```
Uncaught Error: Permission denied to access property "__hb_original"
  instrument honeybadger.js:580
  load honeybadger.js:1716
  load honeybadger.js:1737
  configure honeybadger.js:830
  configure honeybadger.js:830
  configure honeybadger.js:2045
```

We've also seen this error in our own projects—[here's an error report](https://titanic.honeybadger.io/projects/5/faults/33117#notice-context).

## What versions are you using?
Here's a user agent from that titanic error report:

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:108.0) Gecko/20100101 Firefox/108.0
```

That instance was with @honeybadger-io/js 4.8.2. Another customer is using 3.2.8 and has the same error (that's the one that causes their app to crash).

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_henxd)

**Comment by joshuap** [comment]
I've reproduced the issue locally! Here's how:

1. Open http://example.com/ in Firefox

2. Open the console and try this code:
```
window['onpopstate']
```

This is what I see:

<img width="490" alt="image" src="https://user-images.githubusercontent.com/474649/213272950-0e150a09-5320-42fa-ab7c-434a6c481cc9.png">

3. Install the [Readwise Highlighter](https://addons.mozilla.org/en-US/firefox/addon/readwise-highlighter/) firefox addon and reload the page

4. Run this code in the console again:

```
window['onpopstate']
```

<img width="490" alt="image" src="https://user-images.githubusercontent.com/474649/213273066-dd91c5c6-4c8c-42cb-aa90-944f9c8b2fda.png">

5. Finally, try to access `__hb_original` (or any property) on the restricted object:

```
window['onpopstate'].__hb_original
```

<img width="490" alt="image" src="https://user-images.githubusercontent.com/474649/213273260-868aff4c-7631-4b0c-af38-8416aa210a01.png">

I think the problem is that a) the addon must be adding its own version of `onpopstate` (which doesn't normally seem to exist in Firefox—what's up with that?), and b) it's a restricted object because of [permissions related to the same-origin policy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Property_access_denied).

**Comment by subzero10** [comment]
Thanks for the detailed info and instructions to reproduce! I will get right on it.

**Comment by subzero10** [comment]
This should be fixed with v4.9.2!

**Comment by joshuap** [comment]
I have verified that this fixes the bug. Thanks for the quick turnaround!
